visvis2/material/_line.py

```diff
@@ -1,7 +1,27 @@
 from ._base import Material
 
 import python_shader
-from python_shader import vec3, mat4
+from python_shader import vec3, mat4, Array
+
+
+# Idea:
+# scene objects (like stdinfo, lights) are at bind group 0
+# the objects defined by the geometry are at bind group 1
+# the objects defined by the material are at bind group 2
+# that leaves (by default) one bind group to be used for something else
```

Comment:
are there always 4 bind groups?

Reply:
The default number of bind groups (that is requested when you create a device) is 4. You can request more, but then the hardware might say no.